app/util/data_preparation/jsd_prepare_data.py

```diff
@@ -0,0 +1,263 @@
+              f'Retrieved {len(issues_list)} issues')
+        print('---------------------------------------------------')
+        return issues_list
+
+    print(f'Force retrieving {TOTAL_ISSUES_TO_RETRIEVE} issues from Service Desk projects')
+    projects_keys = ','.join([project['projectKey'] for project in service_desks])
+    issues = jira_api.issues_search(jql=f"project in ({projects_keys})", max_results=TOTAL_ISSUES_TO_RETRIEVE)
+    if len(issues) < TOTAL_ISSUES_TO_RETRIEVE:
+        raise Exception(f"ERROR: Jira Service Desks does not have enough issues "
+                        f"{len(issues)}/{TOTAL_ISSUES_TO_RETRIEVE}")
+    issues_list = []
+    for issue in issues:
+        for service_desk in service_desks_issues:
+            if service_desk['projectKey'] in issue['key']:
+                issue_str = f'{issue["id"]},' \
+                            f'{issue["key"]},' \
+                            f'{service_desk["serviceDeskId"]},' \
+                            f'{service_desk["projectId"]},' \
+                            f'{service_desk["projectKey"]}'
+
+                issues_list.append(issue_str)
+    print(f"Force retrieved {len(issues_list)} issues.")
+    print('---------------------------------------------------')
+    return issues_list
+
+
+def generate_random_string(length = 20):
+    return "".join([random.choice(string.ascii_lowercase) for _ in range(length)])
+
+
+def __write_to_file(file_path, items):
+    with open(file_path, 'w') as f:
+        for item in items:
+            f.write(f"{item}\n")
+
+
+def __create_data_set(jira_client, jsd_client):
+    dataset = dict()
+    dataset[AGENTS] = __get_agents(jira_client)
+    dataset[CUSTOMERS] = __get_customers(jira_client, jsd_client)
+    dataset[ISSUES] = __get_requests(jira_api=jira_client, jsd_api=jsd_client)
+    dataset[SERVICE_DESKS] = __get_service_desks(jsd_api=jsd_client)
+
+    return dataset
+
+
+def write_test_data_to_files(datasets):
+    agents = [f"{user['name']},{DEFAULT_PASSWORD}" for user in datasets[AGENTS]]
+    __write_to_file(JSD_DATASET_AGENTS, agents)
+
+    __write_to_file(JSD_DATASET_CUSTOMERS, datasets[CUSTOMERS])
+    __write_to_file(JSD_DATASET_ISSUES, datasets[ISSUES])
+    __write_to_file(JSD_DATASET_SERVICE_DESKS, datasets[SERVICE_DESKS])
+
+
+def main():
+    print("Started preparing data")
+    url = JSD_SETTINGS.server_url
+    print("Server url: ", url)
+
+    jsd_client = JsdRestClient(url, JSD_SETTINGS.admin_login, JSD_SETTINGS.admin_password,
+                               headers=JSD_EXPERIMENTAL_HEADERS)
+    jira_client = JiraRestClient(url, JSD_SETTINGS.admin_login, JSD_SETTINGS.admin_password)
+
+    import time
+    time_now = time.time()
+    dataset = __create_data_set(jira_client=jira_client, jsd_client=jsd_client)
+    write_test_data_to_files(dataset)
+    print("--- %s seconds ---" % (time.time() - time_now))  # TODO delete it after fully developing prepare_data.py
```

Comment:
I would instead propose to add print_time decorator to util functions. Decorator already created in the tooling jsd branch.